flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/exceptionhistory/FailureHandlingResultSnapshotTest.java

```diff
@@ -115,7 +156,12 @@ public void testLocalFailureHandlingResultSnapshotCreation() {
                 FailureHandlingResultSnapshot.create(
                         failureHandlingResult, this::getExecutionVertex);
 
-        assertThat(testInstance.getRootCause(), is(rootCause));
+        // SerializableThrowable needs to be wrapped around to expose the deserializeError method -
+        // the check would fail, otherwise.
+        assertThat(
+                new SerializedThrowable(testInstance.getRootCause())
+                        .deserializeError(ClassLoader.getSystemClassLoader()),
+                is(rootCause));
```

Comment:
The test also passes w/o this wrapping.

Reply:
Good catch. I did some code changes afterwards that made the wrapping obsolete. 👍